schema_salad/avro/schema.py

```diff
@@ -438,6 +505,14 @@ def __init__(
         # Call parent ctor
         Schema.__init__(self, "union")
 
+        # Store name and namespace as they were read in origin schema
+        if name is not None:
+            # Add class members
+            new_name = names.add_name(name, namespace, cast(NamedSchema, self))
```

Comment:
Can't do this, `UnionSchema` is not a subclass of `NamedSchema`

https://github.com/common-workflow-language/schema_salad/actions/runs/4156123750/jobs/7189685177#step:6:139

Reply:
Yes because not all `union` objects have a name. This is why I asked if it was necessary to create a separate  NamedUnionSchema class.